Panic observed at point of peermsg.(*Inv).Decode on invalid wire messages

```
panic: runtime error: makeslice: len out of range

goroutine 7366152 [running]:
github.com/dusk-network/dusk-blockchain/pkg/p2p/peer/peermsg.(*Inv).Decode(0xc00084b9b0, 0xc0018ef500, 0xc7cf763b4989c309, 0x0)
        /root/go/pkg/mod/github.com/dusk-network/dusk-blockchain@v0.2.1-0.20191229115357-479399269943/pkg/p2p/peer/peermsg/inventory.go:63 +0x64
github.com/dusk-network/dusk-blockchain/pkg/p2p/peer/responding.(*DataBroker).SendItems(0xc001702400, 0xc0018ef500, 0x90faa5, 0xc00084ba80)
        /root/go/pkg/mod/github.com/dusk-network/dusk-blockchain@v0.2.1-0.20191229115357-479399269943/pkg/p2p/peer/responding/databroker.go:38 +0x6f
github.com/dusk-network/dusk-blockchain/pkg/p2p/peer.(*messageRouter).route(0xc0007e8ba0, 0xc0018ef506, 0xc0018ef500)
        /root/go/pkg/mod/github.com/dusk-network/dusk-blockchain@v0.2.1-0.20191229115357-479399269943/pkg/p2p/peer/router.go:63 +0x499
github.com/dusk-network/dusk-blockchain/pkg/p2p/peer.(*messageRouter).Collect(0xc0007e8ba0, 0xc0018ef500, 0x4, 0x47b7d1d2e2646101)
        /root/go/pkg/mod/github.com/dusk-network/dusk-blockchain@v0.2.1-0.20191229115357-479399269943/pkg/p2p/peer/router.go:39 +0x7b
github.com/dusk-network/dusk-blockchain/pkg/p2p/peer.(*Reader).ReadLoop(0xc001702420)
        /root/go/pkg/mod/github.com/dusk-network/dusk-blockchain@v0.2.1-0.20191229115357-479399269943/pkg/p2p/peer/peer.go:249 +0xda
created by github.com/dusk-network/dusk-explorer/node.(*Node).OnAccept
        /root/dusk-explorer/node/node.go:170 +0x4f7

```

--- a/pkg/config/loader.go
+++ b/pkg/config/loader.go
@@ -236,4 +236,5 @@ func init() {
 	r.Wallet.Store = "walletDB"
 	r.Consensus.DefaultLockTime = 1000
 	r.Consensus.DefaultAmount = 10
+	r.Mempool.MaxInvItems = 10000
 }

--- a/pkg/core/mempool/mempool_test.go
+++ b/pkg/core/mempool/mempool_test.go
@@ -68,6 +68,7 @@ func TestMain(m *testing.M) {
 	r := config.Registry{}
 	r.Mempool.MaxSizeMB = 1
 	r.Mempool.PoolType = "hashmap"
+	r.Mempool.MaxInvItems = 10000
 	config.Mock(&r)
 
 	var streamer *eventbus.GossipStreamer

--- a/pkg/p2p/peer/peermsg/inventory.go
+++ b/pkg/p2p/peer/peermsg/inventory.go
@@ -2,8 +2,10 @@ package peermsg
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
 
+	"github.com/dusk-network/dusk-blockchain/pkg/config"
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/encoding"
 )
 
@@ -29,6 +31,10 @@ type Inv struct {
 }
 
 func (i *Inv) Encode(w *bytes.Buffer) error {
+	if uint32(len(i.InvList)) > config.Get().Mempool.MaxInvItems {
+		return errors.New("inv message is too large")
+	}
+
 	if err := encoding.WriteVarInt(w, uint64(len(i.InvList))); err != nil {
 		return err
 	}
@@ -60,9 +66,12 @@ func (inv *Inv) Decode(r *bytes.Buffer) error {
 		return err
 	}
 
+	if lenVect > uint64(config.Get().Mempool.MaxInvItems) {
+		return errors.New("inv message is too large")
+	}
+
 	inv.InvList = make([]InvVect, lenVect)
 	for i := uint64(0); i < lenVect; i++ {
-
 		var invType uint8
 		if err := encoding.ReadUint8(r, &invType); err != nil {
 			return err

--- a/pkg/p2p/peer/peermsg/inventory_test.go
+++ b/pkg/p2p/peer/peermsg/inventory_test.go
@@ -2,10 +2,13 @@ package peermsg_test
 
 import (
 	"bytes"
+	"math"
 	"testing"
 
-	crypto "github.com/dusk-network/dusk-crypto/hash"
+	"github.com/dusk-network/dusk-blockchain/pkg/config"
 	"github.com/dusk-network/dusk-blockchain/pkg/p2p/peer/peermsg"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/encoding"
+	crypto "github.com/dusk-network/dusk-crypto/hash"
 	"github.com/stretchr/testify/assert"
 )
 
@@ -24,4 +27,25 @@ func TestEncodeDecodeInventory(t *testing.T) {
 	}
 
 	assert.Equal(t, inv, inv2)
+}
+
+func TestSizeLimit(t *testing.T) {
+	// Encoding
+	hash, _ := crypto.RandEntropy(32)
+	inv := &peermsg.Inv{}
+	for i := 0; uint32(i) < config.Get().Mempool.MaxInvItems+1; i++ {
+		inv.AddItem(peermsg.InvTypeBlock, hash)
+	}
+
+	buf := new(bytes.Buffer)
+	assert.Error(t, inv.Encode(buf))
+
+	// Decoding
+	// We encode an inv message manually, so we dont actually have
+	// to create one with math.MaxUint64 items in it.
+	buf = new(bytes.Buffer)
+	assert.NoError(t, encoding.WriteVarInt(buf, math.MaxUint64))
+
+	inv = &peermsg.Inv{}
+	assert.Error(t, inv.Decode(buf))
 }

